pkg/scheduler/framework/plugins/podtopologyspread/common.go

```diff
@@ -52,7 +52,7 @@ func (tsc *topologySpreadConstraint) matchNodeInclusionPolicies(pod *v1.Pod, nod
 	}
 
 	if tsc.NodeTaintsPolicy == v1.NodeInclusionPolicyHonor {
-		if _, untolerated := v1helper.FindMatchingUntoleratedTaint(node.Spec.Taints, pod.Spec.Tolerations, nil); untolerated {
+		if _, untolerated := v1helper.FindMatchingUntoleratedTaint(node.Spec.Taints, pod.Spec.Tolerations, helper.DoNotScheduleTaintsFilterFunc()); untolerated {
```

Comment:
Thanks, it's reasonable, we should filter out `PreferNoSchedule` ones. But we need to add more tests to cover this, maybe in `TestPreFilterStateAddPod`, `TestPreFilterState` and `TestPodTopologySpreadScore`, not too much, but we should cover them.

Reply:
make sense

Reply:
This change makes sense, but I am looking at the usage of the function `matchNodeInclusionPolicies`; in https://github.com/kubernetes/kubernetes/blob/661899f1a14680840e06c84bc3ff46f73d203012/pkg/scheduler/framework/plugins/podtopologyspread/filtering.go#L210 we are calling the function while checking the nodeAffinity constraints of the preemptor pod, but for taints we are checking the ones of the pod we are adding/removing from the node, this is also not correct. We need to fix that and add unit tests and patch both fixes back.

Reply:
It's a big mistake made by me before, I'll fix it ASAP.

Reply:
Hi @kerthcet

Reply:
Already fixed, see https://github.com/kubernetes/kubernetes/pull/112507, I should broadcast it. Sorry for that.